src/helpers/images/image_provider.ts

```diff
@@ -0,0 +1,59 @@
+import { FigureSize, Getters } from "../../types";
+import { FileUploaderInterface, ImageProviderInterface } from "../../types/files";
+
+export class ImageProvider implements ImageProviderInterface {
+  private fileUploader: FileUploaderInterface;
+  private maxWidth: number;
+  private maxHeight: number;
+
+  constructor(fileUploader: FileUploaderInterface, getter: Getters) {
+    this.fileUploader = fileUploader;
+    const dim = getter.getSheetViewDimension();
+    this.maxWidth = dim.width;
+    this.maxHeight = dim.height;
+  }
+
+  async requestImage(): Promise<{ path: string; size: FigureSize }> {
+    const file = await this.getImageFromUser();
+    const path = await this.fileUploader.upload(file);
```

Comment:
the upload could fail for many reasons.
It would be better to handle errors gracefully.
(probably where `requestImage` is called)